src/manifolds/Sphere.jl

```diff
@@ -290,8 +290,9 @@ log(::AbstractSphere, ::Any...)
 
 function log!(M::AbstractSphere, X, p, q)
     cosθ = real(dot(p, q))
+    cosθ = abs(cosθ) > 1 ? sign(cosθ) * one(cosθ) : cosθ
```

Comment:
Since now you have upper and lower bounds together, we could just use `clamp`:

```suggestion
    cosθ = clamp(real(dot(p, q)), -1, 1)
```

Reply:
> What does this PR change?

It adds the check for -1/clamp for the acos (and/or approx)? At least I ran once into the situation, where the approx did not trigger (let's say for the extreme case of `-1.1`) and I ended up in the second case. So for security I would like to clamp upfront.

I also had checks in the second case, that for `norm(p-q)` small or  `X` small one directly returns the zero-vector, too, but I could not get that stably to run and it does not seem necessary, so only the clamp stayed.

Reply:
(i am not sure why it was added here when I said, quote in reply and it added it here?)

Reply:
> It adds the check for -1/clamp for the acos (and/or approx)? At least I ran once into the situation, where the approx did not trigger (let's say for the extreme case of `-1.1`) and I ended up in the second case. So for security I would like to clamp upfront.

Ah, good point! Okay, this is fine then.